mycroft/configuration/config.py

```diff
@@ -203,20 +227,39 @@ def load_config_stack(configs=None, cache=False):
         Returns:
             (dict) merged dict of all configuration files
         """
+        # system administrators can define different constraints in how
+        # configurations are loaded
         if not configs:
-            configs = [LocalConf(DEFAULT_CONFIG), RemoteConf(),
-                       LocalConf(SYSTEM_CONFIG), LocalConf(USER_CONFIG),
+            configs = [LocalConf(DEFAULT_CONFIG), LocalConf(SYSTEM_CONFIG),
+                       RemoteConf(), LocalConf(USER_CONFIG),
                        Configuration.__patch]
         else:
             # Handle strings in stack
             for index, item in enumerate(configs):
                 if isinstance(item, str):
                     configs[index] = LocalConf(item)
 
+        # Build maintainers and system administrators may prevent users from
+        # modifying specific keys within the configuration.
+        system_config = LocalConf(SYSTEM_CONFIG) or {}
+        remote_config_disabled = system_config.get(
+            'disable_remote_config', False)
+        user_config_disabled = system_config.get(
+            'disable_user_config', False)
+        protected_keys = system_config.get('protected_keys', [])
+
         # Merge all configs into one
         base = {}
-        for c in configs:
-            merge_dict(base, c)
+        for config in configs:
+
+            # handle system constraints
+            if isinstance(config, RemoteConf) or \
+               isinstance(config, LocalConf) and config.path == USER_CONFIG:
+                if user_config_disabled:
```

Comment:
shouldn't this account for `remote_config_disabled`?

Reply:
That is a pretty important piece of the puzzle yes... thanks!

Added.